### PR TITLE
Fix segmentation fault

### PR DESCRIPTION
**Pull request by snehaa8**

### CODE REVIEW

File: utilities/test_suite/HIP/Tensor_hip.cpp [hunk 1]
```diff
@@ -1093,7 +1093,7 @@ int main(int argc, char **argv)
                     2.input bit depth 0 (U8)
                     3.source and destination layout are the same*/
                     if(qaFlag && inputBitDepth == 0 && (srcDescPtr->layout == dstDescPtr->layout) && !(randomOutputCase))
-                        compare_reduction_output(static_cast<Rpp64u *>(reductionFuncResultArr), testCaseName, srcDescPtr, testCase, dst, scriptPath);
+                        compare_reduction_output(static_cast<uint64_t *>(reductionFuncResultArr), testCaseName, srcDescPtr, testCase, dst, scriptPath);
```
Comment (r-abishek) [inline utilities/test_suite/HIP/Tensor_hip.cpp:1096]: Why the change from Rpp64u to uint64_t?

**Comment by r-abishek** [comment]
Was the error shown as a Segmentation Fault or a Double Free? Is there a screenshot?

**Review by r-abishek — APPROVED** [review]

**Comment by snehaa8** [comment]
I saw a "corrupted size vs. prev_size" error

File: utilities/test_suite/rpp_test_suite_common.h [hunk 2]
```diff
@@ -1111,21 +1113,25 @@ inline void compare_reduction_output(Rpp64u* output, string funcName, RpptDescPt
             func += "Tensor_PLN1";
     }
 
-    refFile = scriptPath + "/../REFERENCE_OUTPUT/" + funcName + "/"+ binFile + ".bin";
-
-    string line,word;
-    int index = 0;
     int fileMatch = 0;
     int matched_values = 0;
-    Rpp64u *binaryContent = (Rpp64u *)malloc(binaryOutputSize * sizeof(Rpp64u));
```
[the remark this answers is not in the document]
Comment (snehaa8) [reply utilities/test_suite/rpp_test_suite_common.h:1120]: I am not very sure
However other image kernels to use a similar malloc.

File: utilities/test_suite/HIP/Tensor_hip.cpp (hunk 1 above)
Comment (snehaa8) [reply utilities/test_suite/HIP/Tensor_hip.cpp:1096]: I had initially missed using abs while comparing present run outputs with reference outputs. 
I got build error

"call to 'abs' is ambiguous"
            int diff = abs(output[i] - refOutput[i]);
Using uint64_t seemed to fix the error
